docs/development/compiler/README.md

```diff
@@ -1,3 +1,130 @@
-# The Qooxdoo compiler
+# The Qooxdoo Compiler and Command Line Toolchain
 
-TBD.
+Starting with v6, Qooxdoo includes a brand new command line interface called `qx` which automates many of the 
+routine tasks involved in creating and managing your application code, including compiling for development or 
+release onto production servers.
+
+Using `qx` you quickly and easily perform daily routine tasks, such as:
+* Compile your code!
+* Serve your code via a built in web server
+* Install packages written by yourself or others
+* Create skeleton applications
+
+For many people, the easiest way to get started with Qooxdoo is to actually use *just* the `qx` command, rather than
+try and download Qooxdoo itself - once you have installed the `qx` command, it knows how to download, install, and
+manage Qooxdoo.  
+
+But it's not just Qooxdoo that it can download - if someone writes a reusable "package" (which is 
+just a library of code and resources), `qx` can find it on GitHub and then will be able to download and install that
+package too.  This is really useful because packages can have dependencies on other packages, so letting `qx` do
+the hard work of finding and downloading things isn't just useful for getting started - it can be a crucial tool
+in getting reliable compilation, whether that means getting it working for everyone in your team, or just making 
+sure that your project is up to date with the latest version of all it's dependencies.
+
+## Installation / Getting Started
+
+The `qx` command is easily installed with `npm` and as it is written in 100% Javascript it is entirely cross 
+platform; the only requirement is that you use a currently supported version of NodeJs itself (well - we do *try* to 
+support older versions of NodeJs, but it's not guaranteed).
+
+For many people, the easiest way to do this is to install `qx` globally - installing globally is a great solution
+if you just want the easy simplicity for getting started with Qooxdoo, or knowing that all your projects are always
+on the latest version of Qooxdoo and the compiler.
+
+To install globally, all you need to do is this:
+
+```
+npm i -g @qooxdoo/compiler
+```
+
+### Creating your first application
+
+Let's create a "skeleton" application and then compile it:
+
+```
+qx create myFirstApp --noninteractive
+cd myFirstApp
+qx compile
+```
+
+Now use your favourite web browser to open the `compiled/source/index.html` - that's it, your first
+Qooxdoo application is working!
+
+(BTW did you see that first command had `--noninteractive`?  That just means to take the defaults; if you miss it off, then
```

Comment:
"miss" may be a British usage here, but would be used in American usage. Maybe `if you leave it off` ?